lib/ansible/playbook/base.py

```diff
@@ -231,12 +233,37 @@ def load_data(self, ds, variable_manager=None, loader=None):
                 else:
                     self._attributes[target_name] = ds[name]
 
+        # set defaults from CLI
+        if options:
+            self._set_options(options)
+
```

Comment:
This looks very very fragile.  In addition to what alikins said earlier, we're taking an arbitrary data structure and overwriting attributes in the dictionary if they exist.  If they don't exist, then we're discarding them.  This removes the programming language's builtin abilities to tell when bad data is being passed to it.

This also doesn't seem like it is an inherent part of a become plugin so even if we want to try it, it should be part of a different PR.

Reply:
its would be a prerequisite, since we need a way to get options into the task object from which become consumes.

Reply:
We should be sending specific options in or (better design) the object should be requesting the defaults that it needs.

Reply:
i agree, but for this change i just copied what we do in play_context, I would leave redoing the API for a subsequent PR ... probably after we remove play_context from the picture.

Reply:
It probably should go into a PR before this one.

The easiest thing to do is probably to save the CLI options to a program global once it is parsed.  We currently use ansible.constants in that manner so ansible.constants.cli_options might be a suitable name.